### PR TITLE
=art add metadata field to the header, #20703

### PR DESCRIPTION
**Pull request by patriknw**
incorporating feedback from https://github.com/akka/akka/pull/21378

Don't squash with github UI. I'll do that so that Konrad gets the attribution.

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #20703

**Comment by patriknw** [comment]
@ktoso @bantonsson Incorporated the final review feedback here: fa0f5b9

I have created a separate ticket for the ByteBuffer: https://github.com/akka/akka/issues/21482

File: akka-remote/src/main/scala/akka/remote/artery/RemoteInstrument.scala [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  def create(system: ExtendedActorSystem): Vector[RemoteInstrument] = {
+    val c = system.settings.config
+    val path = "akka.remote.artery.advanced.instruments"
+    import scala.collection.JavaConverters._
+    c.getStringList(path).asScala.map { fqcn ⇒
+      system
+        .dynamicAccess.createInstanceFor[RemoteInstrument](fqcn, Nil)
+        .orElse(system.dynamicAccess.createInstanceFor[RemoteInstrument](fqcn, List(classOf[ExtendedActorSystem] → system)))
+        .get
+    }(collection.breakOut)
```
Comment (bantonsson) [inline akka-remote/src/main/scala/akka/remote/artery/RemoteInstrument.scala]: Nice! I always thing of Breakout the game when I read this 😉

**Review by bantonsson — APPROVED** [review]
LGTM

**Pull request merged**